pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -659,6 +679,59 @@ func (e *BuildExpression) updateRequirement(requirement model.Requirement) error
 	return nil
 }
 
+func (e *BuildExpression) UpdatePlatform(operation model.Operation, platformID strfmt.UUID, timestamp strfmt.DateTime) error {
+	var err error
+	switch operation {
+	case model.OperationAdded:
+		err = e.addPlatform(platformID)
+	case model.OperationRemoved:
+		err = e.removePlatform(platformID)
+	default:
+		return errs.New("Unsupported operation")
+	}
+	if err != nil {
+		return errs.Wrap(err, "Could not update BuildExpression's platform")
+	}
+
+	err = e.updateTimestamp(timestamp)
+	if err != nil {
+		return errs.Wrap(err, "Could not update BuildExpression's timestamp")
+	}
```

Comment:
This doesn't feel like it should be part of UpdatePlatform. It should have its own call.